walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,62 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //some basic if statements
+    if(request.getAttendees().isEmpty()){
+        //when there are no attendees
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    } else if(request.getDuration() > TimeRange.END_OF_DAY) {
+        //when the meeting duration is longer than a day
+        return Arrays.asList();
+    }
+    
+    //list of TimeRanges that will not work for meeting attendees
+    Collection<TimeRange> blockedTimes = new ArrayList();
+
+    //traverse through all attendees that need to attend the meeting
+    for(String attendee: request.getAttendees()) {
+        //traverse through all of the events occuring
+        for(Event event: events){
+            //check if that attendee is attending that event
+            if(event.getAttendees().contains(attendee)) {
+                //must block off that TimeRange
+                blockedTimes.add(event.getWhen());
+            }
+        }
+    }
```

Comment:
same thing here. just add a comment for the entire for block. you don't need to have one for every line, most people will understand

like what you did for `//in ascending order, add the available TimeRanges into the availableTimes list `

Reply:
Fixed, thanks!